src/web/parserUtils.ts

```diff
@@ -179,19 +179,22 @@ function parseText(
 
   const markdownRanges: MarkdownRange[] = ranges as MarkdownRange[];
 
-  targetElement.innerHTML = '';
-  targetElement.innerText = '';
-
   // We don't want to parse text with single '\n', because contentEditable represents it as invisible <br />
   if (!!text && text !== '\n') {
     const dom = parseRangesToHTMLNodes(text, markdownRanges, markdownStyle);
-    target.appendChild(dom);
-  }
 
-  if (alwaysMoveCursorToTheEnd) {
-    CursorUtils.moveCursorToEnd(target);
-  } else if (isFocused && cursorPosition !== null) {
-    CursorUtils.setCursorPosition(target, cursorPosition, disableNewLinesInCursorPositioning);
+    // eslint-disable-next-line es/no-optional-chaining
+    if ((targetElement.firstChild as HTMLElement | null)?.innerHTML !== dom.innerHTML) {
```

Comment:
I'm not a fan of disabling eslint rules, what about changing it to something like this:
```suggestion
if (!targetElement.firstChild || (targetElement.firstChild as HTMLElement).innerHTML !== dom.innerHTML) {
```

Reply:
Also, could you please explain why do we need to access `.firstChild` of `targetElement` and not the `targetElement` itself?

Reply:
This is because the `target` element is the div itself, so if we try retrieving it's `innerHTML` it will include root span, whereas if we retrieve `innerHTML` from `dom` it won't be including the root span @tomekzaw